src/oca_github_bot/webhooks/on_pr_update_check_cla.py

```diff
@@ -0,0 +1,383 @@
+        """
+        res = self.db.execute(self.SQL_SELECT_MISS_CACHE_NOLOGIN_PR, (self.pr_key,))
+        notified_miss = {r[0] for r in res.fetchall()}
+        remaining_no_login = {
+            (name, email)
+            for name, email in users_no_login
+            if email not in notified_miss
+        }
+        self._update_cache_miss_no_login(remaining_no_login)
+        return remaining_no_login
+
+    def _update_cache_miss_no_login(self, users_no_login):
+        sql = (
+            "INSERT OR IGNORE "
+            "INTO nologin_cla_ko(email, date_cla_ko, pull_request) "
+            "VALUES(?, CURRENT_DATE, ?)"
+        )
+        pr_key = self.pr_key
+        params = [(email, pr_key) for name, email in users_no_login]
+        self.db.executemany(sql, params)
+
+
+def get_commit_author(commit):
+    """
+    Check a commit from the github api json
+    """
+    message = commit["commit"]["message"]
+    if re.match(RE_WEBLATE_COMMIT_MSG, message, re.S):
+        # don't enforce CLA check on translations
+        return set(), set()
+    users_login = set()
+    users_no_login = set()
+    if commit["committer"] and "login" in commit["committer"]:
+        author = commit["committer"]["login"]
+        users_login.add(author)
+    else:
+        author = commit["commit"]["committer"]["name"]
+        author_email = commit["commit"]["committer"]["email"]
+        users_no_login.add((author, author_email))
+    if commit["author"] and "login" in commit["author"]:
+        author = commit["author"]["login"]
+        users_login.add(author)
+    else:
+        author = commit["commit"]["author"]["name"]
+        author_email = commit["commit"]["author"]["email"]
+        users_no_login.add((author, author_email))
+
+    return users_login, users_no_login
+
+
+SQL_DATABASE_CREATE = """
+    CREATE TABLE IF NOT EXISTS login_cla_ok(
+        login VARCHAR(80),
+        date_cla_ok TEXT
+    );
+    CREATE UNIQUE INDEX IF NOT EXISTS login_cla_ok_unique
+    ON login_cla_ok(login);
+    CREATE INDEX IF NOT EXISTS login_cla_ok_index
+    ON login_cla_ok(login);
+
+    CREATE TABLE IF NOT EXISTS login_cla_ko(
+        login VARCHAR(80),
+        date_cla_ko TEXT,
+        pull_request TEXT
+    );
+    CREATE UNIQUE INDEX IF NOT EXISTS login_cla_ko_unique
+    ON login_cla_ko(login, pull_request);
+    CREATE INDEX IF NOT EXISTS login_cla_ko_index
+    ON login_cla_ko(login, pull_request);
+
+    CREATE TABLE IF NOT EXISTS nologin_cla_ko(
+        name VARCHAR(80),
+        email VARCHAR(80),
+        date_cla_ko TEXT,
+        pull_request TEXT
+    );
+    CREATE UNIQUE INDEX IF NOT EXISTS nologin_cla_ko_unique
+    ON nologin_cla_ko(email, pull_request);
+    CREATE INDEX IF NOT EXISTS nologin_cla_ko_index
+    ON nologin_cla_ko(email, pull_request);
+
+"""
+
+
+def _init_database(dbfile):
+    with sqlite3.connect(dbfile) as db:
+        db.executescript(SQL_DATABASE_CREATE)
+
+
+_init_database(config.CLABOT_CACHE)
```

Comment:
@sbidoul I'm not quite sure how to handle this initialization. Do you have a suggestion?

Reply:
[main](https://github.com/OCA/oca-github-bot/blob/master/src/oca_github_bot/__main__.py) could call various initializers.

Or run it at the beginning of the task if the database does not exist?